File: android-exercises/3 - intent/Intent_java/app/src/main/java/com/bugrahankaramollaoglu/intent_java/MainActivity2.java
package com.bugrahankaramollaoglu.intent_java;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity2 extends AppCompatActivity {

    private String receivedName;
    private String receivedAge;
    private Button changeBackButton;
    private TextView receivedNameText;
    private TextView receivedAgeText;
    private Intent intent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);

        changeBackButton = findViewById(R.id.button2);
        receivedNameText = findViewById(R.id.receivedNameText);
        receivedAgeText = findViewById(R.id.receivedAgeText);

        // intentle A2'de verileri alırken ilklendirmeyi unutma, yoksa cöküyor
        intent = getIntent();

        // bu anahtarlarla vermiştik, yine aynı anahtarla aliyoruz
        receivedName = intent.getStringExtra("namekey");
        receivedAge = intent.getStringExtra("agekey");


        receivedNameText.setText("received name: " + receivedName);
        receivedAgeText.setText("received age: " + receivedAge);
    }

    public void change2(View view) {
        Intent intent = new Intent(this, MainActivity.class);
        startActivity(intent);
    }
}

File: android-exercises/3 - intent/Intent_kotlin/app/src/main/java/com/bugrahankaramollaoglu/intent_kotlin/MainActivity2.kt
package com.bugrahankaramollaoglu.intent_kotlin


import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity2 : AppCompatActivity() {

    private lateinit var intent: Intent
    private lateinit var changeBack: Button
    private lateinit var nameView: TextView
    private lateinit var ageView: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)

        changeBack = findViewById<Button>(R.id.changeBackButton)
        nameView = findViewById<TextView>(R.id.nameTextView)
        ageView = findViewById<TextView>(R.id.ageTextView)

        intent = getIntent()

        val name: String? = intent.getStringExtra("namekey")
        val age: String? = intent.getStringExtra("agekey")

        nameView.text = "your name: ${name}"
        ageView.text = "your name: ${age}"



        changeBack.setOnClickListener {
            val intent = Intent(this@MainActivity2, MainActivity::class.java)
            startActivity(intent)
        }
    }


}

